webapp/components/channel_invite_button.jsx

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+
+import * as AsyncClient from 'utils/async_client.jsx';
+import * as Client from 'utils/client.jsx';
+
+import {FormattedMessage} from 'react-intl';
+
+import loadingGif from 'images/load.gif';
+
+export default class ChannelInviteButton extends React.Component {
```

Comment:
I wonder if this should be `SpinnerButton` so that it can be re-usable within other components. Could just have the onClick passed in like a normal button, etc.

Maybe not worth doing in this PR because it looks like it might be a decent amount of refactoring, but probably worth a ticket